src/Authenticator/TokenAuthenticator.php

```diff
@@ -118,4 +119,14 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
 
         return new Result($user, Result::SUCCESS);
     }
+
+    /**
+     * No-op method.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request A request object.
+     * @return void
+     */
+    public function unauthorizedChallenge(ServerRequestInterface $request)
+    {
```

Comment:
Token authenticator probably should be stateless. It would be used in APIs mostly.
Or we could have another `StatelessTokenAuthenticator` so a developer could decide what they want.

BTW is JWT authentication always stateless?